wait-for-idle-runner/main.js

```diff
@@ -0,0 +1,35 @@
+const { Octokit } = require("@octokit/action");
+
+const octokit = new Octokit();
+
+const NUMBER_OF_ATTEMPTS = 3000;
+const TIME_BETWEEN_ATTEMPTS_SECONDS = 5000;
+
+async function main() {
+  for (var i=0; i < NUMBER_OF_ATTEMPTS; i++) {
+    const result = await octokit.request(`GET /repos/${process.env.REPOSITORY_NAME}/actions/runners`)
+
+    // Select the runner based on its name
+    runner = result.data["runners"].filter(function(runner) {
+      return runner.name == process.env.RUNNER_NAME;
+    })[0]
+
+    if (typeof runner == "undefined") {
+      console.log("::set-output name=runner-found::false");
```

Comment:
I don't think any changes are needed but I'm just curious if there's a default for `runner-idle` in this case. I think (although I could be wrong) that it defaults to some type of empty value that's treated as false?

It probably doesn't matter because the `runner-found` value should probably be checked before even bothering with `runner-idle`

Reply:
`runner-found` needs to be checked first, so I did not add any `runner-idle` for this case. Not sure what happens with the value but it is probably set to nothing. I would not worry for this case too much right now, and we can improve is this is needed. The `runner-found` check is an edge case check I added because it was asked for in one of the reviews: initially I did not even plan for the runner to not exist, as it should always be there.